Error using v 0.1

Hi,

Nice project! 

I was trying to use it into my environment and I'm facing the following error:

```
/usr/local/bin/kubectl get sa --all-namespaces --output json
/usr/local/bin/kubectl get roles --all-namespaces --output json
/usr/local/bin/kubectl get rolebindings --all-namespaces --output json
/usr/local/bin/kubectl get clusterroles --output json
/usr/local/bin/kubectl get clusterrolebindings --output json
panic: interface conversion: interface {} is nil, not []interface {}

goroutine 1 [running]:
main.lookupRoles(0xc0000a1650, 0x7, 0xc0000a1640, 0x7, 0xc000060150, 0xc000192270, 0xc0000ea400, 0x1c, 0x20, 0xc00023eb40, ...)
	/Users/hausenbl/go/src/github.com/mhausenblas/rback/main.go:199 +0x586
main.genGraph(0xc000060150, 0xc000192270, 0xc0000ea400, 0x1c, 0x20, 0xc00023eb40, 0xc0000eae00, 0x1c, 0x20, 0x0)
	/Users/hausenbl/go/src/github.com/mhausenblas/rback/main.go:310 +0xc01
main.main()
	/Users/hausenbl/go/src/github.com/mhausenblas/rback/main.go:27 +0x141

```

My env:

OS CentOS 7.6 64 bit
K8S version: 1.13.4

Tks

--- a/main.go
+++ b/main.go
@@ -159,27 +159,27 @@ func getPermissions() (Permissions, error) {
 	p := Permissions{}
 	sa, err := getServiceAccounts()
 	if err != nil {
- 		return p, err
+		return p, err
 	}
-	p.ServiceAccounts = sa 
+	p.ServiceAccounts = sa
 	roles, err := getRoles()
 	if err != nil {
- 		return p, err
+		return p, err
 	}
 	p.Roles = roles
 	rb, err := getRoleBindings()
 	if err != nil {
- 		return p, err
+		return p, err
 	}
 	p.RoleBindings = rb
 	cr, err := getClusterRoles()
 	if err != nil {
- 		return p, err
+		return p, err
 	}
 	p.ClusterRoles = cr
 	crb, err := getClusterRoleBindings()
 	if err != nil {
- 		return p, err
+		return p, err
 	}
 	p.ClusterRoleBindings = crb
 	return p, nil
@@ -196,11 +196,13 @@ func lookupRoles(namespace, sa string, p Permissions) (roles []string, err error
 		}
 		roleRef := d["roleRef"].(map[string]interface{})
 		r := roleRef["name"].(string)
-		subjects := d["subjects"].([]interface{})
-		for _, subject := range subjects {
-			s := subject.(map[string]interface{})
-			if s["name"] == sa {
-				roles = append(roles, r)
+		if d["subjects"] != nil {
+			subjects := d["subjects"].([]interface{})
+			for _, subject := range subjects {
+				s := subject.(map[string]interface{})
+				if s["name"] == sa {
+					roles = append(roles, r)
+				}
 			}
 		}
 	}
@@ -218,13 +220,16 @@ func lookupClusterRoles(sa string, p Permissions) (clusterroles []string, err er
 		}
 		roleRef := d["roleRef"].(map[string]interface{})
 		r := roleRef["name"].(string)
-		subjects := d["subjects"].([]interface{})
-		for _, subject := range subjects {
-			s := subject.(map[string]interface{})
-			if s["name"] == sa {
-				clusterroles = append(clusterroles, r)
+		if d["subjects"] != nil {
+			subjects := d["subjects"].([]interface{})
+			for _, subject := range subjects {
+				s := subject.(map[string]interface{})
+				if s["name"] == sa {
+					clusterroles = append(clusterroles, r)
+				}
 			}
 		}
+
 	}
 	return clusterroles, nil
 }
@@ -233,7 +238,7 @@ func lookupClusterRoles(sa string, p Permissions) (clusterroles []string, err er
 // if namespace is empty then the scope is cluster-wide.
 func lookupResources(namespace, role string, p Permissions) (resources string, err error) {
 	if namespace != "" { // look up in roles
-		for _, roles := range p.Roles[namespace] {	
+		for _, roles := range p.Roles[namespace] {
 			var d map[string]interface{}
 			b := []byte(roles)
 			err = json.Unmarshal(b, &d)
@@ -253,7 +258,7 @@ func lookupResources(namespace, role string, p Permissions) (resources string, e
 		}
 	}
 	// ... otherwise, look up in cluster roles:
-	for _, cr := range p.ClusterRoles {	
+	for _, cr := range p.ClusterRoles {
 		var d map[string]interface{}
 		b := []byte(cr)
 		err = json.Unmarshal(b, &d)
@@ -330,12 +335,12 @@ func genGraph(p Permissions) *dot.Graph {
 	}
 	return g
 }
- 
+
 // struct2json turns a map into a JSON string
-func struct2json(s map[string]interface{})(string, error){
+func struct2json(s map[string]interface{}) (string, error) {
 	str, err := json.Marshal(s)
 	if err != nil {
-	    return "", err
+		return "", err
 	}
 	return string(str), nil
 }

